### PR TITLE
minimal namespace rules for modules / services

### PR DESCRIPTION
**Pull request by alex40724**
We recently had a discussion about how to move forward with namespaces in the TB. This PR is a suggested next step.

Any feedback is appreciated.

### CODE REVIEW

File: docs/development/namespaces.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Namespaces
+
+ILIAS is using PSR-4 namespaces in its [src directory](../../src/README.md).
+
+You MAY also use namespaces in the `Modules` or `Services` directories. You SHOULD use `ILIAS\ModuleName` or `ILIAS\ServiceName`, e.g. `ILIAS\Course` or `ILIAS\ActiveRecord` in these cases.
+
+* Please be informed that we might introduce a PSR-4 based autoloading for these components in the future, too, and revise our directory structure accordingly. **You MAY only make use of these namespaces if you are willing to support these necessary future refactorings**.
```
Comment (chfsx) [inline docs/development/namespaces.md:7]: @alex40724 thank you very much for the proposal! Just one question: Is `ilCtrl` already ready to work with namespaces? (this is really a question, I just don't know)

**Review by chfsx — APPROVED** [review]

**Review by Amstutz — APPROVED** [review]
thx.

File: docs/development/namespaces.md (hunk 1 above)
Comment (alex40724) [reply docs/development/namespaces.md:7]: @chfsx Thanks, this is an important question.

No, ilCtrl is not ready (yet). I think there are several places that are not ready. Also the ilObject... related factories and any other place that works "generic factory" like.

But we are witnessing several places where namespaces are already used, e.g. https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/Modules/OrgUnit/PublicApi/OrgUnitUserService.php or in Services/FileDelivery. And we would like to allow these approaches by introducing this small guidline.

**Review by kergomard — APPROVED** [review]
Thanks!

**Review by mjansenDatabay — APPROVED** [review]
:shipit:

**Comment by matthiaskunkel** [comment]
Jour Fixe, 29 JUL 2019: We highly appreciate this suggestion and accept it for trunk. Please merge this PR.

File: docs/development/namespaces.md (hunk 1 above)
Comment (ghost) [reply docs/development/namespaces.md:7]: I listed some limitations or bugs with namespace guis in ilCtrl in https://mantis.ilias.de/view.php?id=26181

**Pull request merged**